protocols/v2/binary-sv2/no-serde-sv2/codec/src/datatypes/non_copy_data_types/inner.rs

```diff
@@ -1,3 +1,33 @@
+/// Provides a flexible container for managing either owned or mutable references to byte arrays.
+///
+/// # Overview
+/// Defines the `Inner` enum to manage both mutable references to byte slices and owned vectors (`Vec<u8>`). 
+/// Accommodates both fixed-size and variable-size data using const generics, offering control over size and header length constraints.
+///
+/// # `Inner` Enum
+/// The `Inner` enum has two variants for data management:
+/// - `Ref(&'a mut [u8])`: A mutable reference to a byte slice, allowing in-place data modification.
+/// - `Owned(Vec<u8>)`: An owned byte vector, providing full control over data and supporting move semantics.
+///
+/// ## Const Parameters
+/// Configured using const generics for the following constraints:
+/// - `ISFIXED`: Indicates whether the data has a fixed size.
+/// - `SIZE`: Specifies the size when `ISFIXED` is true.
+/// - `HEADERSIZE`: Defines the size of the header, useful for variable-size data with a prefix length.
+/// - `MAXSIZE`: Limits the maximum allowable size of the data.
+///
+/// # Usage
+/// `Inner` offers several methods for data manipulation, including:
+/// - `to_vec()`: Returns a `Vec<u8>`, cloning the slice or owned data.
+/// - `inner_as_ref()` and `inner_as_mut()`: Provide immutable or mutable access to the data.
+/// - `expected_length(data: &[u8])`: Computes the expected length, validating it against constraints.
+/// - `get_header()`: Returns the data's header based on `HEADERSIZE`.
+///
+/// # Implementations
+/// The `Inner` enum implements `PartialEq`, `Eq`, `GetSize`, `SizeHint`, and `Sv2DataType` traits, enabling buffer size calculations, reading, and writing to byte slices.
+///
+/// # Error Handling
+/// Methods return `Error` types when data exceeds size limits or deviates from the configuration, ensuring compliance with defined constraints.
```

Comment:
I cannot find a rendering of this header anywhere in the Rust Docs when I do `cargo doc`

no point in using `///` here